src/generator/operations.ts

```diff
@@ -427,10 +427,15 @@ function createProtocolRequest(client: string, op: Operation, imports: ImportMan
   const mediaType = getMediaType(op.requests![0].protocol);
   if (mediaType === 'JSON' || mediaType === 'XML') {
     const bodyParam = values(aggregateParameters(op)).where((each: Parameter) => { return each.protocol.http!.in === 'body'; }).first();
+    // adding this variable to control whether a 'options.' needs to be added before optional body parameters
+    let setOptionsPrefix = false;
```

Comment:
I think this var can be removed, then down on line 478 say `if (!bodyParam!.required)`

Reply:
Oh hmm maybe not, since it's not set to `true` in certain cases.